### ISSUE TITLE
jdk12.0.1+0-preview_openj9-0.15.0-m1 s390x crash running test MathLoadTest_bigdecimal_ConcurrentScavenge

### ISSUE THREAD

**Issue by lumpfish** [eclipse/openj9#6411]
Test failure at AdoptOpenJDK:

Failure link
------------
Jenkins job output has been cleaned, some captured output of the crash is below.

Test category: system
Test case: MathLoadTest_bigdecimal_ConcurrentScavenge (openjdk-systemtest repo)
OS/architecture: Linux s390x
OpenJDK version:
```
10:49:05  Run /data/jenkins/workspace/Grinder/openjdkbinary/j2sdk-image/bin/java -version
10:49:10  openjdk version "12.0.1" 2019-04-16
10:49:10  OpenJDK Runtime Environment AdoptOpenJDK (build 12.0.1+0)
10:49:10  Eclipse OpenJ9 VM AdoptOpenJDK (build openj9-0.15.0-m1, JRE 12 Linux s390x-64-Bit Compressed References 20190704_135 (JIT enabled, AOT enabled)
10:49:10  OpenJ9   - 655fac772
10:49:10  OMR      - dc661ac3
10:49:10  JCL      - e8d0f8b8d4 based on jdk-12.0.2+0+preview)
```

Optional info
-------------

- intermittent failure (yes|no):  yes - test passed on rerun

Failure output
--------------
00:06:09  MLT stderr Unhandled exception
00:06:09  MLT stderr Type=Segmentation error vmState=0x00000000
00:06:09  MLT stderr Unhandled exception
00:06:09  MLT stderr Type=Segmentation error vmState=0x00000000
00:06:09  MLT stderr J9Generic_Signal_Number=00000018 Signal_Number=0000000b Error_Value=c7533e10 Signal_Code=00000001
00:06:09  MLT stderr Handler1=000003FFA74C8DD0 Handler2=000003FFA72A7CE0 InaccessibleAddress=00000001954B7000
00:06:09  MLT stderr J9Generic_Signal_Number=00000018 Signal_Number=0000000b Error_Value=34163e10 Signal_Code=00000001
00:06:09  MLT stderr Handler1=000003FFA74C8DD0 Handler2=000003FFA72A7CE0 InaccessibleAddress=00000001954B7000
00:06:09  MLT stderr gpr0=000000000000000A gpr1=0000000000000017 gpr2=000000010F770CD0 gpr3=0000000000000001
00:06:09  MLT stderr gpr4=00000000000A7200 gpr5=000000000049F9A0 gpr6=000000010F770CDC gpr7=0000000100000034
00:06:09  MLT stderr gpr8=0000000000000100 gpr9=0000000000000001 gpr10=00000000954B7760 gpr11=000000010000000A
00:06:09  MLT stderr gpr12=0000000100000002 gpr13=0000000000466A00 gpr14=000003FF86F0F2A0 gpr15=000003FFA6E7DD48
00:06:09  MLT stderr gpr0=000000000000000A gpr1=0000000000000013 gpr2=000000010F622E88 gpr3=0000000000000002
00:06:09  MLT stderr gpr4=00000000000A7200 gpr5=0000000000470DC0 gpr6=000000010F622E94 gpr7=0000000100000032
00:06:09  MLT stderr gpr8=0000000000000100 gpr9=0000000000000002 gpr10=00000000954B7760 gpr11=000000010000000A
00:06:09  MLT stderr gpr12=0000000100000002 gpr13=000000000046B400 gpr14=000003FF86F0F2A0 gpr15=000003FFA6D7E4A8
00:06:09  MLT stderr psw=000003FF86FEC118 mask=0705100180000000 fpc=0038ff00 bea=000003FF86FEC0DA
00:06:09  MLT stderr psw=000003FF86FEC118 mask=0705100180000000 fpc=0038ff00 bea=000003FF86FEC0DA
00:06:09  MLT stderr fpr0 000000000000babe (f: 47806.000000, d: 2.361930e-319)
00:06:09  MLT stderr fpr1 0020bab80000f401 (f: 62465.000000, d: 4.653011e-308)
00:06:09  MLT stderr fpr2 0000004800000041 (f: 65.000000, d: 1.527837e-312)
00:06:09  MLT stderr fpr3 bff0000000000000 (f: 0.000000, d: -1.000000e+00)
00:06:09  MLT stderr fpr4 0000000000000000 (f: 0.000000, d: 0.000000e+00)
00:06:09  MLT stderr fpr5 0000000000000000 (f: 0.000000, d: 0.000000e+00)
00:06:09  MLT stderr fpr6 0000000000000000 (f: 0.000000, d: 0.000000e+00)
00:06:09  MLT stderr fpr7 3c61b00000000000 (f: 0.000000, d: 7.670730e-18)
00:06:09  MLT stderr fpr8 40734413509f79fe (f: 1352628736.000000, d: 3.082547e+02)
00:06:09  MLT stderr fpr9 000003ffa036ab10 (f: 2687937280.000000, d: 2.172130e-311)
00:06:09  MLT stderr fpr10 000003ffa6e40000 (f: 2799960064.000000, d: 2.172185e-311)
00:06:09  MLT stderr fpr11 000003ffe10ff713 (f: 3775919872.000000, d: 2.172667e-311)
00:06:09  MLT stderr fpr12 00000000000fb648 (f: 1029704.000000, d: 5.087414e-318)
00:06:09  MLT stderr fpr13 000003ffc137e058 (f: 3241664512.000000, d: 2.172403e-311)
00:06:09  MLT stderr fpr14 0000000000000022 (f: 34.000000, d: 1.679823e-322)
00:06:09  MLT stderr fpr15 000003ff8187ec98 (f: 2173168896.000000, d: 2.171875e-311)
00:06:09  MLT stderr 
00:06:09  MLT stderr Compiled_method=java/math/BigInteger.<init>([CII)V
00:06:09  MLT stderr Target=2_90_20190704_135 (Linux 4.4.0-116-generic)
00:06:09  MLT stderr CPU=s390x (4 logical CPUs) (0x1e9a20000 RAM)
00:06:09  MLT stderr ----------- Stack Backtrace -----------
00:06:09  MLT stderr fpr0 0000000095878378 (f: 2508686080.000000, d: 1.239456e-314)
00:06:09  MLT stderr fpr1 0020bab80000f401 (f: 62465.000000, d: 4.653011e-308)
00:06:09  MLT stderr fpr2 0000004800000041 (f: 65.000000, d: 1.527837e-312)
00:06:09  MLT stderr fpr3 4040aaaaaaaaaaab (f: 2863311616.000000, d: 3.333333e+01)
00:06:09  MLT stderr fpr4 483fa24080000000 (f: 2147483648.000000, d: 1.076442e+40)
00:06:09  MLT stderr fpr5 3d7548de57e64617 (f: 1474708992.000000, d: 1.209892e-12)
00:06:09  MLT stderr fpr6 0000000000000000 (f: 0.000000, d: 0.000000e+00)
00:06:09  MLT stderr fpr7 bc5c000000000000 (f: 0.000000, d: -6.071532e-18)
00:06:09  MLT stderr fpr8 40734413509f79fe (f: 1352628736.000000, d: 3.082547e+02)
00:06:09  MLT stderr fpr9 000003ffa036ab10 (f: 2687937280.000000, d: 2.172130e-311)
00:06:09  MLT stderr fpr10 000003ffa6d40000 (f: 2798911488.000000, d: 2.172185e-311)
00:06:09  MLT stderr fpr11 000003ffe10ff713 (f: 3775919872.000000, d: 2.172667e-311)
00:06:09  MLT stderr fpr12 00000000000fb648 (f: 1029704.000000, d: 5.087414e-318)
00:06:09  MLT stderr fpr13 000003ffc137e058 (f: 3241664512.000000, d: 2.172403e-311)
00:06:09  MLT stderr fpr14 0000000000000022 (f: 34.000000, d: 1.679823e-322)
00:06:09  MLT stderr fpr15 000003ff8187ec98 (f: 2173168896.000000, d: 2.171875e-311)
00:06:09  MLT stderr 
00:06:09  MLT stderr Compiled_method=java/math/BigInteger.<init>([CII)V
00:06:09  MLT stderr Target=2_90_20190704_135 (Linux 4.4.0-116-generic)
00:06:09  MLT stderr CPU=s390x (4 logical CPUs) (0x1e9a20000 RAM)
00:06:09  MLT stderr ----------- Stack Backtrace -----------
00:06:09  MLT stderr Unhandled exception
00:06:09  MLT stderr Type=Segmentation error vmState=0x00000000
00:06:09  MLT stderr J9Generic_Signal_Number=00000018 Signal_Number=0000000b Error_Value=84bbfc68 Signal_Code=00000001
00:06:09  MLT stderr Handler1=000003FFA74C8DD0 Handler2=000003FFA72A7CE0 InaccessibleAddress=00000001954B7000
00:06:09  MLT stderr gpr0=0000000000000012 gpr1=0000000000000144 gpr2=000000010F7AA4C8 gpr3=0000000000000003
00:06:09  MLT stderr gpr4=00000000000A7200 gpr5=000000000047E660 gpr6=000000010F7AA4E4 gpr7=0000000100000030
00:06:09  MLT stderr gpr8=0000000000000100 gpr9=0000000000000003 gpr10=00000000954B7760 gpr11=0000000100000012
00:06:09  MLT stderr gpr12=000000010000000A gpr13=0000000000468F00 gpr14=000003FF86F0F2A0 gpr15=000003FFA6DFDD48
00:06:09  MLT stderr psw=000003FF86FEC118 mask=0705100180000000 fpc=0038ff00 bea=000003FF86FEC0DA
00:06:09  MLT stderr (0x000003FF86FEC118 [<unknown>+0x0])
00:06:09  MLT stderr ---------------------------------------
00:06:09  MLT stderr fpr0 000000000000babe (f: 47806.000000, d: 2.361930e-319)
00:06:09  MLT stderr fpr1 483fa240bbc58836 (f: 3150284800.000000, d: 1.076442e+40)
00:06:09  MLT stderr fpr2 0000000000000000 (f: 0.000000, d: 0.000000e+00)
00:06:09  MLT stderr fpr3 3fb002ba2eb367ff (f: 783509504.000000, d: 6.254161e-02)
00:06:09  MLT stderr fpr4 483fa24080000000 (f: 2147483648.000000, d: 1.076442e+40)
00:06:09  MLT stderr fpr5 3d75476749a49af6 (f: 1235524352.000000, d: 1.209567e-12)
00:06:09  MLT stderr fpr6 0000000000000000 (f: 0.000000, d: 0.000000e+00)
00:06:09  MLT stderr fpr7 3c4c000000000000 (f: 0.000000, d: 3.035766e-18)
00:06:09  MLT stderr fpr8 406ff262073071c9 (f: 120615368.000000, d: 2.555745e+02)
00:06:09  MLT stderr fpr9 000003ffa036ab10 (f: 2687937280.000000, d: 2.172130e-311)
00:06:09  MLT stderr fpr10 000003ffa6dc0000 (f: 2799435776.000000, d: 2.172185e-311)
00:06:09  MLT stderr fpr11 000003ffe10ff713 (f: 3775919872.000000, d: 2.172667e-311)
00:06:09  MLT stderr fpr12 00000000000fb648 (f: 1029704.000000, d: 5.087414e-318)
00:06:09  MLT stderr fpr13 000003ffc137e058 (f: 3241664512.000000, d: 2.172403e-311)
00:06:09  MLT stderr fpr14 0000000000000022 (f: 34.000000, d: 1.679823e-322)
00:06:09  MLT stderr fpr15 000003ff8187ec98 (f: 2173168896.000000, d: 2.171875e-311)
00:06:09  MLT stderr 
00:06:09  MLT stderr Compiled_method=java/math/BigInteger.<init>([CII)V
00:06:09  MLT stderr Target=2_90_20190704_135 (Linux 4.4.0-116-generic)
00:06:09  MLT stderr CPU=s390x (4 logical CPUs) (0x1e9a20000 RAM)
00:06:09  MLT stderr ----------- Stack Backtrace -----------
00:06:09  MLT stderr (0x000003FF86FEC118 [<unknown>+0x0])
00:06:09  MLT stderr ---------------------------------------
00:06:09  MLT stderr (0x000003FF86FEC118 [<unknown>+0x0])
00:06:09  MLT stderr ---------------------------------------
00:06:09  MLT stderr JVMDUMP039I Processing dump event "gpf", detail "" at 2019/07/04 19:06:08 - please wait.
00:06:09  MLT stderr JVMDUMP039I Processing dump event "gpf", detail "" at 2019/07/04 19:06:08 - please wait.
00:06:09  MLT stderr JVMDUMP039I Processing dump event "gpf", detail "" at 2019/07/04 19:06:08 - please wait.
00:06:09  MLT stderr JVMDUMP032I JVM requested System dump using '/data/jenkins/workspace/openjdk12_j9_systemtest_s390x_linux/openjdk-tests/TestConfig/test_output_15622739486928/MathLoadTest_bigdecimal_ConcurrentScavenge_0/20190704-190531-MathLoadTest/results/core.20190704.190608.32759.0001.dmp' in response to an event
00:06:10  MLT stderr JVMPORT030W /proc/sys/kernel/core_pattern setting "|/usr/share/apport/apport %p %s %c %d %P" specifies that the core dump is to be piped to an external program.  Attempting to rename either core or core.32982.
00:06:10  MLT stderr 
00:06:10  MLT stderr JVMDUMP010I System dump written to /data/jenkins/workspace/openjdk12_j9_systemtest_s390x_linux/openjdk-tests/TestConfig/test_output_15622739486928/MathLoadTest_bigdecimal_ConcurrentScavenge_0/20190704-190531-MathLoadTest/results/core.20190704.190608.32759.0001.dmp
00:06:10  MLT stderr JVMDUMP032I JVM requested Java dump using '/data/jenkins/workspace/openjdk12_j9_systemtest_s390x_linux/openjdk-tests/TestConfig/test_output_15622739486928/MathLoadTest_bigdecimal_ConcurrentScavenge_0/20190704-190531-MathLoadTest/results/javacore.20190704.190608.32759.0004.txt' in response to an event
00:06:10  MLT stderr JVMDUMP032I JVM requested System dump using '/data/jenkins/workspace/openjdk12_j9_systemtest_s390x_linux/openjdk-tests/TestConfig/test_output_15622739486928/MathLoadTest_bigdecimal_ConcurrentScavenge_0/20190704-190531-MathLoadTest/results/core.20190704.190608.32759.0002.dmp' in response to an event
00:06:10  STF 19:06:10.195 - Found dump at: /data/jenkins/workspace/openjdk12_j9_systemtest_s390x_linux/openjdk-tests/TestConfig/test_output_15622739486928/MathLoadTest_bigdecimal_ConcurrentScavenge_0/20190704-190531-MathLoadTest/results/javacore.20190704.190608.32759.0004.txt
00:06:10  STF 19:06:10.195 - Found dump at: /data/jenkins/workspace/openjdk12_j9_systemtest_s390x_linux/openjdk-tests/TestConfig/test_output_15622739486928/MathLoadTest_bigdecimal_ConcurrentScavenge_0/20190704-190531-MathLoadTest/results/core.20190704.190608.32759.0001.dmp
00:06:11  MLT stderr javacore file generated - /data/jenkins/workspace/openjdk12_j9_systemtest_s390x_linux/openjdk-tests/TestConfig/test_output_15622739486928/MathLoadTest_bigdecimal_ConcurrentScavenge_0/20190704-190531-MathLoadTest/results/javacore.20190704.190608.32759.0004.txt
00:06:11  MLT stderr core file generated - /data/jenkins/workspace/openjdk12_j9_systemtest_s390x_linux/openjdk-tests/TestConfig/test_output_15622739486928/MathLoadTest_bigdecimal_ConcurrentScavenge_0/20190704-190531-MathLoadTest/results/core.20190704.190608.32759.0001.dmp
00:06:33  MLT stderr JVMPORT030W /proc/sys/kernel/core_pattern setting "|/usr/share/apport/apport %p %s %c %d %P" specifies that the core dump is to be piped to an external program.  Attempting to rename either core or core.32997.
00:06:33  MLT stderr 
00:06:33  MLT stderr JVMDUMP010I System dump written to /data/jenkins/workspace/openjdk12_j9_systemtest_s390x_linux/openjdk-tests/TestConfig/test_output_15622739486928/MathLoadTest_bigdecimal_ConcurrentScavenge_0/20190704-190531-MathLoadTest/results/core.20190704.190608.32759.0002.dmp
00:06:33  MLT stderr JVMDUMP032I JVM requested Java dump using '/data/jenkins/workspace/openjdk12_j9_systemtest_s390x_linux/openjdk-tests/TestConfig/test_output_15622739486928/MathLoadTest_bigdecimal_ConcurrentScavenge_0/20190704-190531-MathLoadTest/results/javacore.20190704.190608.32759.0005.txt' in response to an event
00:06:33  MLT stderr JVMDUMP032I JVM requested System dump using '/data/jenkins/workspace/openjdk12_j9_systemtest_s390x_linux/openjdk-tests/TestConfig/test_output_15622739486928/MathLoadTest_bigdecimal_ConcurrentScavenge_0/20190704-190531-MathLoadTest/results/core.20190704.190608.32759.0003.dmp' in response to an event
00:06:33  STF 19:06:31.629 - Found dump at: /data/jenkins/workspace/openjdk12_j9_systemtest_s390x_linux/openjdk-tests/TestConfig/test_output_15622739486928/MathLoadTest_bigdecimal_ConcurrentScavenge_0/20190704-190531-MathLoadTest/results/javacore.20190704.190608.32759.0005.txt
00:06:33  STF 19:06:31.630 - Found dump at: /data/jenkins/workspace/openjdk12_j9_systemtest_s390x_linux/openjdk-tests/TestConfig/test_output_15622739486928/MathLoadTest_bigdecimal_ConcurrentScavenge_0/20190704-190531-MathLoadTest/results/core.20190704.190608.32759.0002.dmp
00:06:33  MLT stderr javacore file generated - /data/jenkins/workspace/openjdk12_j9_systemtest_s390x_linux/openjdk-tests/TestConfig/test_output_15622739486928/MathLoadTest_bigdecimal_ConcurrentScavenge_0/20190704-190531-MathLoadTest/results/javacore.20190704.190608.32759.0005.txt
00:06:33  MLT stderr core file generated - /data/jenkins/workspace/openjdk12_j9_systemtest_s390x_linux/openjdk-tests/TestConfig/test_output_15622739486928/MathLoadTest_bigdecimal_ConcurrentScavenge_0/20190704-190531-MathLoadTest/results/core.20190704.190608.32759.0002.dmp
00:07:00  MLT stderr JVMPORT030W /proc/sys/kernel/core_pattern setting "|/usr/share/apport/apport %p %s %c %d %P" specifies that the core dump is to be piped to an external program.  Attempting to rename either core or core.33113.
00:07:00  MLT stderr 
00:07:00  MLT stderr JVMDUMP010I System dump written to /data/jenkins/workspace/openjdk12_j9_systemtest_s390x_linux/openjdk-tests/TestConfig/test_output_15622739486928/MathLoadTest_bigdecimal_ConcurrentScavenge_0/20190704-190531-MathLoadTest/results/core.20190704.190608.32759.0003.dmp
00:07:00  MLT stderr JVMDUMP032I JVM requested Java dump using '/data/jenkins/workspace/openjdk12_j9_systemtest_s390x_linux/openjdk-tests/TestConfig/test_output_15622739486928/MathLoadTest_bigdecimal_ConcurrentScavenge_0/20190704-190531-MathLoadTest/results/javacore.20190704.190608.32759.0006.txt' in response to an event
00:07:00  STF 19:06:57.664 - Found dump at: /data/jenkins/workspace/openjdk12_j9_systemtest_s390x_linux/openjdk-tests/TestConfig/test_output_15622739486928/MathLoadTest_bigdecimal_ConcurrentScavenge_0/20190704-190531-MathLoadTest/results/javacore.20190704.190608.32759.0006.txt
00:07:00  STF 19:06:57.665 - Found dump at: /data/jenkins/workspace/openjdk12_j9_systemtest_s390x_linux/openjdk-tests/TestConfig/test_output_15622739486928/MathLoadTest_bigdecimal_ConcurrentScavenge_0/20190704-190531-MathLoadTest/results/core.20190704.190608.32759.0003.dmp
00:07:00  MLT stderr javacore file generated - /data/jenkins/workspace/openjdk12_j9_systemtest_s390x_linux/openjdk-tests/TestConfig/test_output_15622739486928/MathLoadTest_bigdecimal_ConcurrentScavenge_0/20190704-190531-MathLoadTest/results/javacore.20190704.190608.32759.0006.txt
00:07:00  MLT stderr core file generated - /data/jenkins/workspace/openjdk12_j9_systemtest_s390x_linux/openjdk-tests/TestConfig/test_output_15622739486928/MathLoadTest_bigdecimal_ConcurrentScavenge_0/20190704-190531-MathLoadTest/results/core.20190704.190608.32759.0003.dmp
00:07:52  MLT stderr JVMDUMP010I Java dump written to /data/jenkins/workspace/openjdk12_j9_systemtest_s390x_linux/openjdk-tests/TestConfig/test_output_15622739486928/MathLoadTest_bigdecimal_ConcurrentScavenge_0/20190704-190531-MathLoadTest/results/javacore.20190704.190608.32759.0004.txt
00:07:52  MLT stderr JVMDUMP010I Java dump written to /data/jenkins/workspace/openjdk12_j9_systemtest_s390x_linux/openjdk-tests/TestConfig/test_output_15622739486928/MathLoadTest_bigdecimal_ConcurrentScavenge_0/20190704-190531-MathLoadTest/results/javacore.20190704.190608.32759.0005.txt
00:07:52  MLT stderr JVMDUMP032I JVM requested Snap dump using '/data/jenkins/workspace/openjdk12_j9_systemtest_s390x_linux/openjdk-tests/TestConfig/test_output_15622739486928/MathLoadTest_bigdecimal_ConcurrentScavenge_0/20190704-190531-MathLoadTest/results/Snap.20190704.190608.32759.0007.trc' in response to an event
00:07:52  MLT stderr JVMDUMP010I Java dump written to /data/jenkins/workspace/openjdk12_j9_systemtest_s390x_linux/openjdk-tests/TestConfig/test_output_15622739486928/MathLoadTest_bigdecimal_ConcurrentScavenge_0/20190704-190531-MathLoadTest/results/javacore.20190704.190608.32759.0006.txt
00:07:52  MLT stderr JVMDUMP010I Snap dump written to /data/jenkins/workspace/openjdk12_j9_systemtest_s390x_linux/openjdk-tests/TestConfig/test_output_15622739486928/MathLoadTest_bigdecimal_ConcurrentScavenge_0/20190704-190531-MathLoadTest/results/Snap.20190704.190608.32759.0007.trc
00:07:52  MLT stderr JVMDUMP007I JVM Requesting JIT dump using '/data/jenkins/workspace/openjdk12_j9_systemtest_s390x_linux/openjdk-tests/TestConfig/test_output_15622739486928/MathLoadTest_bigdecimal_ConcurrentScavenge_0/20190704-190531-MathLoadTest/results/jitdump.20190704.190608.32759.0010.dmp'
00:07:52  MLT stderr JVMDUMP010I JIT dump written to /data/jenkins/workspace/openjdk12_j9_systemtest_s390x_linux/openjdk-tests/TestConfig/test_output_15622739486928/MathLoadTest_bigdecimal_ConcurrentScavenge_0/20190704-190531-MathLoadTest/results/jitdump.20190704.190608.32759.0010.dmp
00:07:52  MLT stderr JVMDUMP013I Processed dump event "gpf", detail "".
00:07:52  STF 19:07:45.741 - Found dump at: /data/jenkins/workspace/openjdk12_j9_systemtest_s390x_linux/openjdk-tests/TestConfig/test_output_15622739486928/MathLoadTest_bigdecimal_ConcurrentScavenge_0/20190704-190531-MathLoadTest/results/Snap.20190704.190608.32759.0007.trc
00:07:52  STF 19:07:45.741 - **FAILED** Process MLT ended with exit code (255) and not the expected exit code/s (0)
00:07:52  MLT stderr Snap file generated - /data/jenkins/workspace/openjdk12_j9_systemtest_s390x_linux/openjdk-tests/TestConfig/test_output_15622739486928/MathLoadTest_bigdecimal_ConcurrentScavenge_0/20190704-190531-MathLoadTest/results/Snap.20190704.190608.32759.0007.trc
00:07:52  STF 19:07:45.741 - Monitoring Report Summary:
00:07:52  STF 19:07:45.741 -   o Process MLT has crashed unexpectedly
00:07:52  STF 19:07:45.741 - Killing processes: MLT
00:07:52  STF 19:07:45.741 -   o Process MLT is not running
00:07:52  **FAILED** at step 1 (Run math load test). Expected return value=0 Actual=1 at /data/jenkins/workspace/openjdk12_j9_systemtest_s390x_linux/openjdk-tests/TestConfig/scripts/testKitGen/../../../TestConfig/test_output_15622739486928/MathLoadTest_bigdecimal_ConcurrentScavenge_0/20190704-190531-MathLoadTest/execute.pl line 94.
00:07:52  STF 19:07:45.793 - **FAILED** execute script failed. Expected return value=0 Actual=1
00:07:52  STF 19:07:45.793 -

**Comment by lumpfish** [comment]
Failure reoccurred once in a 10x Grinder rerun: https://ci.adoptopenjdk.net/view/Test_grinder/job/Grinder-systemtest/176/consoleFull

**Comment by DanHeidinga** [comment]
Crash in a compiled method:
```Compiled_method=java/math/BigInteger.([CII)V```

@fjeremic Can you take a look at this?  Note, it's a crash on the `0.15.0` milestone 1 build so we'll need to determine if this has to be fixed before the release asap.

**Comment by DanHeidinga** [comment]
Tentatively tagging this against the 0.15 milestone while the determine of whether or not it's a stop ship.

**Comment by lumpfish** [comment]
The dumps from the 10x Grinder job are in the job output:  https://ci.adoptopenjdk.net/view/Test_grinder/job/Grinder-systemtest/176/artifact/systemtest_test_output.tar.gz

**Comment by fjeremic** [comment]
@NigelYiboYu can you help take a look? I'm fairly certain this is software CS.

**Comment by pshipton** [comment]
@fjeremic is this a dup of https://github.com/eclipse/openj9/issues/6248 ?

**Comment by fjeremic** [comment]
> @fjeremic is this a dup of #6248 ?

Good point. Thanks for the suggestion. I looked at the core file from https://github.com/eclipse/openj9/issues/6411#issuecomment-510049872 and we can see:

```
(kca) (0x000003FF8D36AD18-32)/12i
0x3ff8d36acf8 {java/math/BigInteger.<init>} +2056          0000         Invalid Instruction
0x3ff8d36acfa {java/math/BigInteger.<init>} +2058          0000         Invalid Instruction
0x3ff8d36acfc {java/math/BigInteger.<init>} +2060          47000000     nop       4
0x3ff8d36ad00 {java/math/BigInteger.<init>} +2064          E37060000095 llh       r7,0(,r6) <<< +2008 ^+2130
0x3ff8d36ad06 {java/math/BigInteger.<init>} +2070          C27F00000100 clfi      r7,0x100
0x3ff8d36ad0c {java/math/BigInteger.<init>} +2076          A7A40122     bnlrc     *+580 (0x3FF8D36AF50) C>> +2656
0x3ff8d36ad10 {java/math/BigInteger.<init>} +2080          1839         lr        r3,r9
0x3ff8d36ad12 {java/math/BigInteger.<init>} +2082          1800         lr        r0,r0
0x3ff8d36ad14 {java/math/BigInteger.<init>} +2084          B973A078     clrt      r7,r8,B'1010'
0x3ff8d36ad18 {java/math/BigInteger.<init>} +2088          E307A0080076 lb        r0,8(r7,r10)
0x3ff8d36ad1e {java/math/BigInteger.<init>} +2094          1200         ltr       r0,r0
0x3ff8d36ad20 {java/math/BigInteger.<init>} +2096          A744106A     brl       *+8404 (0x3FF8D36CDF4) C>> +10500
```

This is identical to the instruction sequence seen in https://github.com/eclipse/openj9/issues/6248#issuecomment-505588626. 

The register context says the value of GPR7 has some garbage in the upper half which is the result of the same bug as seen in #6248:

```
08:12:25  MLT stderr Type=Segmentation error vmState=0x00000000
08:12:25  MLT stderr J9Generic_Signal_Number=00000018 Signal_Number=0000000b Error_Value=00000000 Signal_Code=00000001
08:12:25  MLT stderr Handler1=000003FFACB48DD0 Handler2=000003FFAC927CE0 InaccessibleAddress=0000000198B87000
08:12:25  MLT stderr gpr0=000000000000000D gpr1=00000000000002F5 gpr2=0000000113097AD0 gpr3=0000000000000006
08:12:25  MLT stderr gpr4=00000000000A7200 gpr5=000000000046CBC0 gpr6=0000000113097AE2 gpr7=0000000100000035
08:12:25  MLT stderr gpr8=0000000000000100 gpr9=0000000000000006 gpr10=0000000098B87F10 gpr11=000000010000000D
08:12:25  MLT stderr gpr12=0000000000000005 gpr13=0000000000467400 gpr14=000003FF8D5BA29C gpr15=000003FFAC3FDD48
08:12:25  MLT stderr psw=000003FF8D36AD18 mask=0705100180000000 fpc=0038ff00 bea=000003FF8D36ACC8
```

This issue will be resolved once eclipse/omr#4103 is merged and propagated.

**Comment by DanHeidinga** [comment]
@fjeremic Should  eclipse/omr#4103 be ported to the 0.15.0 release branch?  How risky is the change?

**Comment by fjeremic** [comment]
> @fjeremic Should [eclipse/omr#4103](https://github.com/eclipse/omr/pull/4103) be ported to the 0.15.0 release branch? How risky is the change?

The original Load Extensions optimization was introduced as part of eclipse/omr#2205 which was merged on January 13, 2018. The bug has been present in the code base since. We've only in the last few weeks started seeing reproductions of it. I would classify the fix as low risk given we're disabling (removed altogether) a faulty optimization. Since we've started to hit the problem more often now I'm inclined to fix it in 0.15.0 as well. I'll open the PR shortly.
